compiler/src/checker/checker.rs

```diff
@@ -0,0 +1,220 @@
+use crate::CompilerError;
+use crate::CompilerErrorKind;
+use crate::ScopeContainer;
+
+use whistle_ast::IdentType;
+use whistle_ast::Literal;
+use whistle_ast::Operator;
+use whistle_ast::Primitive;
+
+pub struct Checker {
```

Comment:
Would it make sense to break this out to its own crate and also the CompileScope to the common/a new crate?

Reply:
Considering it's not a part of the codegen of the compiler anymore

Reply:
Still todo but in a future pr